### PR TITLE
Example in Java

### PR DESCRIPTION
**Pull request by tpasternak**

### CODE REVIEW

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala [hunk 1]
```diff
@@ -4,18 +4,22 @@ import java.util.concurrent.{SynchronousQueue, ThreadPoolExecutor, TimeUnit}
 import scala.concurrent.ExecutionContext
 import scala.util.Try
 
-trait IdeProbeFixture {
-  protected implicit val ec: ExecutionContext = {
+object IdeProbeFixture {
+  def defaultEc(): ExecutionContext = {
```
Comment (PawelLipski) [inline core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala:8]: Nit:
```suggestion
  def defaultEC(): ExecutionContext = {
```
as per Java conventions for <3 letter acronyms

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala [hunk 2]
```diff
@@ -4,18 +4,22 @@ import java.util.concurrent.{SynchronousQueue, ThreadPoolExecutor, TimeUnit}
...
     val service = new ThreadPoolExecutor(0, Integer.MAX_VALUE, 10L, TimeUnit.SECONDS, new SynchronousQueue[Runnable])
     service.allowCoreThreadTimeOut(true)
```
Comment (PawelLipski) [inline core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala:10]: Would be nice to document why this is needed while you're here before this knowledge is lost **f o r e v e r**

File: core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        var workspaceProvider = new FromResource("gradle-project");
+        var fixture = new IntelliJFixture(
+                workspaceProvider,
+                IntelliJProvider.Default(),
+                Config.Empty(),
+                // API issue - need to use scalaconverters and scala's Function2
+                collectionAsScalaIterable(new ArrayList<Function2<IntelliJFixture, Path, BoxedUnit>>()).toSeq(),
+                collectionAsScalaIterable(new ArrayList<Function2<IntelliJFixture, InstalledIntelliJ, BoxedUnit>>()).toSeq(),
+                collectionAsScalaIterable(new ArrayList<Function2<IntelliJFixture, RunningIntelliJFixture, BoxedUnit>>()).toSeq(),
+                IdeProbeFixture$.MODULE$.defaultEc());
```
Comment (PawelLipski) [inline core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java:37]: Huh... `IdeProbeFixture.defaultEc()`? IIRC Scala compiler creates such static method wrappers for methods of `object`s

File: core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+        fixture.run().apply(intelliJ -> {
+            var expectedProjectName = "foo";
+            var projectRef = intelliJ.probe().openProject(intelliJ.workspace(), WaitLogic.Default()); // API issue - need to pass default WaitLogic
```
Comment (PawelLipski) [inline core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java:41]: Maybe overload `openProject` while you're here?

File: core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java (hunk 3 above)
Comment (tpasternak) [reply core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java]: Done

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala (hunk 2 above)
Comment (tpasternak) [reply core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala:10]: Done

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala (hunk 1 above)
Comment (tpasternak) [reply core/driver/sources/src/main/scala/org/virtuslab/ideprobe/IdeProbeFixture.scala]: Done

File: core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java (hunk 4 above)
Comment (tpasternak) [reply core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java:41]: Yep, I'm going to do it in the next PR, and not only openProject

File: core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+            // API issue: need to use JavaConverters
+            var actualModules = seqAsJavaList(intelliJ.probe().projectModel(projectRef).modules())
+                    .stream()
+                    .map(Module::name)
+                    .collect(toList());
+            var expectedModules = asList("foo.main", "foo.test", "foo.uiTest", "foo");
+            assertEquals(expectedModules, actualModules);
+
+            return null; //API issue - need to return null
```
[the remark this answers is not in the document]
Comment (tpasternak) [reply core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java:60]: It java you see it as `Function1<RunningIntelliJFixture, Object>`

File: core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+public class ProbeDriverJavaTest {
+    @Test
+    public void openProject() {
+        var workspaceProvider = new FromResource("gradle-project");
+        var fixture = new IntelliJFixture(
+                workspaceProvider,
+                IntelliJProvider.Default(),
+                Config.Empty(),
+                // API issue - need to use scalaconverters and scala's Function2
```
Comment (PawelLipski) [inline core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java:33]: Actually... consider opening issues for such problems with Java API, I might delegate someone from git-machete-intellij-plugin to your team, they'll need such entry-level issues :)

Comment (PawelLipski) [reply core/driver/tests/src/test/java/org/virtuslab/ideprobe/ProbeDriverJavaTest.java:33]: Plus reference them here as `// TODO (#1234): ...`

**Review by PawelLipski — APPROVED** [review]
Approving... albeit my approval is powerless here :/

**Pull request merged**